File: vps_hansal_report.py
# Run on the VPS from the tryvera folder:  python vps_hansal_report.py
import sqlite3, json, re, glob, pathlib, collections, datetime
DB = 'data/application_logs.db'
c = sqlite3.connect(DB)
rows = c.execute("select created_at, data from applications where resume_profile='hansal_maniar'").fetchall()
def week(d):
    dt = datetime.date.fromisoformat(d[:10]); return (dt - datetime.timedelta(days=dt.weekday())).isoformat()
gen = collections.Counter(week(r[0]) for r in rows)
print('HANSAL RESUMES GENERATED ON VPS, BY WEEK')
for w in sorted(gen): print('  %s  %4d' % (w, gen[w]))
print('  TOTAL %d' % sum(gen.values()))
# reasoning effort actually used, per week, from the verbose logs
eff = {}
for p in glob.glob('data/logs/verbose/app_*.log'):
    t = pathlib.Path(p).read_text(encoding='utf8', errors='replace')
    r = [int(x) for x in re.findall(r'"reasoning_tokens": (\d+)', t)]
    cost = sum(float(x) for x in re.findall(r'\n  "cost": ([\d.]+)', t))
    eff[pathlib.Path(p).stem] = (max(r) if r else 0, cost)
byweek = collections.defaultdict(lambda: [0, 0, 0.0])
for created, data in rows:
    j = json.loads(data); k = j.get('id', '')
    if k in eff:
        e = byweek[week(created)]; e[0] += 1; e[1] += eff[k][0]; e[2] += eff[k][1]
print('\nEFFORT AND COST BY WEEK (from OpenRouter usage in the logs)')
for w in sorted(byweek):
    n, r, cst = byweek[w]
    level = 'minimal' if r/n < 200 else 'low' if r/n < 3000 else 'medium' if r/n < 20000 else 'high'
    print('  %s  runs %4d  avg reasoning %6d (%s)  cost $%.2f' % (w, n, r/n, level, cst))
# fingerprints so interview PDFs can be traced back to a generation date
fp = {}
for f in glob.glob('output/*/**/result.json', recursive=True):
    try: r = json.loads(pathlib.Path(f).read_text(encoding='utf8'))
    except Exception: continue
    if r.get('resume_profile') != 'hansal_maniar': continue
    md = r.get('resume_markdown') or ''
    bl = [re.sub(r'\W+', '', l[2:])[:80] for l in md.split('\n') if l.startswith('- ')]
    if bl: fp[r.get('created_at', '')[:10] + '|' + (r.get('company_name') or '?')[:30]] = bl
pathlib.Path('hansal_fingerprints.json').write_text(json.dumps(fp), encoding='utf8')
print('\nwrote hansal_fingerprints.json (%d resumes) - send this file back' % len(fp))

File: vps_hansal_report.mjs
// Run on the VPS from the tryvera folder:  node vps_hansal_report.mjs
// Uses node:sqlite (built into the Node the app already runs on), so nothing to install.
import { DatabaseSync } from "node:sqlite";
import fs from "node:fs";
import path from "node:path";

const db = new DatabaseSync("data/application_logs.db", { readOnly: true });
const rows = db.prepare("select created_at, data from applications where resume_profile = 'hansal_maniar'").all();

const week = (d) => {
  const dt = new Date(d.slice(0, 10) + "T00:00:00Z");
  const day = (dt.getUTCDay() + 6) % 7; // Monday = 0
  dt.setUTCDate(dt.getUTCDate() - day);
  return dt.toISOString().slice(0, 10);
};

// 1. Resumes generated per week
const gen = {};
for (const r of rows) gen[week(r.created_at)] = (gen[week(r.created_at)] ?? 0) + 1;
console.log("HANSAL RESUMES GENERATED ON VPS, BY WEEK");
for (const w of Object.keys(gen).sort()) console.log(`  ${w}  ${String(gen[w]).padStart(4)}`);
console.log(`  TOTAL ${rows.length}`);

// 2. Reasoning effort and cost actually used, from OpenRouter usage written into each verbose log
const logDir = "data/logs/verbose";
const eff = {};
if (fs.existsSync(logDir)) {
  for (const f of fs.readdirSync(logDir)) {
    if (!f.startsWith("app_") || !f.endsWith(".log")) continue;
    const t = fs.readFileSync(path.join(logDir, f), "utf8");
    const reas = [...t.matchAll(/"reasoning_tokens": (\d+)/g)].map((m) => +m[1]);
    const cost = [...t.matchAll(/\n  "cost": ([\d.]+)/g)].reduce((s, m) => s + +m[1], 0);
    eff[f.slice(0, -4)] = [reas.length ? Math.max(...reas) : 0, cost];
  }
}
const byWeek = {};
for (const r of rows) {
  const id = JSON.parse(r.data).id ?? "";
  if (!eff[id]) continue;
  const w = week(r.created_at);
  byWeek[w] ??= [0, 0, 0];
  byWeek[w][0] += 1; byWeek[w][1] += eff[id][0]; byWeek[w][2] += eff[id][1];
}
console.log("\nEFFORT AND COST BY WEEK (from OpenRouter usage in the logs)");
for (const w of Object.keys(byWeek).sort()) {
  const [n, reas, cost] = byWeek[w];
  const avg = reas / n;
  const level = avg < 200 ? "minimal" : avg < 3000 ? "low" : avg < 20000 ? "medium" : "high";
  console.log(`  ${w}  runs ${String(n).padStart(4)}  avg reasoning ${String(Math.round(avg)).padStart(6)} (${level})  cost $${cost.toFixed(2)}`);
}

// 3. Fingerprints so interview PDFs can be traced back to the day they were generated
const fp = {};
const walk = (dir) => {
  for (const e of fs.readdirSync(dir, { withFileTypes: true })) {
    const p = path.join(dir, e.name);
    if (e.isDirectory()) walk(p);
    else if (e.name === "result.json") {
      try {
        const r = JSON.parse(fs.readFileSync(p, "utf8"));
        if (r.resume_profile !== "hansal_maniar") continue;
        const bl = (r.resume_markdown ?? "").split("\n").filter((l) => l.startsWith("- "))
          .map((l) => l.slice(2).replace(/\W+/g, "").slice(0, 80));
        if (bl.length) fp[`${(r.created_at ?? "").slice(0, 10)}|${(r.company_name ?? "?").slice(0, 30)}`] = bl;
      } catch { /* unreadable result.json, skip */ }
    }
  }
};
if (fs.existsSync("output")) walk("output");
fs.writeFileSync("hansal_fingerprints.json", JSON.stringify(fp));
console.log(`\nwrote hansal_fingerprints.json (${Object.keys(fp).length} resumes) - send this file back`);
